service/src/main/java/uk/gov/hmcts/reform/fpl/model/UndeliveredEmail.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.hmcts.reform.fpl.model;
+
+import lombok.Builder;
+import lombok.Value;
+import uk.gov.service.notify.Notification;
+
+@Value
+@Builder
+public class UndeliveredEmail {
+    String recipient;
+    String subject;
+    String reference;
+
+    public static UndeliveredEmail fromNotification(Notification notification) {
+        if (notification.getEmailAddress().isEmpty()) {
```

Comment:
could use notification.getEmailAddress().map().orElseThrow(...) to remove the sonar bugs

Reply:
done